pkg/sql/randgen/schema.go

```diff
@@ -186,6 +187,103 @@ func RandCreateTableWithColumnIndexNumberGenerator(
 	return res[0].(*tree.CreateTable)
 }
 
+func parseCreateStatement(createStmtSQL string) (*tree.CreateTable, error) {
+	var p parser.Parser
+	stmts, err := p.Parse(createStmtSQL)
+	if err != nil {
+		return nil, err
+	}
+	if len(stmts) != 1 {
+		return nil, fmt.Errorf("parsed CreateStatement string yielded more than one parsed statment")
+	}
+
+	tableStmt, ok := stmts[0].AST.(*tree.CreateTable)
+	if !ok {
+		return nil, fmt.Errorf("AST could not be cast to *tree.CreateTable")
+	}
+	return tableStmt, nil
+}
+
+// PopulateRandTable populates the provided table with `numrows` rows of random data.
+func PopulateRandTable(rng *rand.Rand, db *gosql.DB, tableName string, numRows int) error {
+	var ignored, createStmtSQL string
+	res := db.QueryRow(fmt.Sprintf("SHOW CREATE TABLE %s", tableName))
+	err := res.Scan(&ignored, &createStmtSQL)
+	if err != nil {
+		return errors.Wrapf(err, "table does not exist in db")
+	}
+	createStmt, err := parseCreateStatement(createStmtSQL)
+	if err != nil {
+		return errors.Wrapf(err, "unable to parse table's create statment")
+	}
+
+	defs := createStmt.Defs
+	colTypes := make([]*types.T, 0)
+	nullable := make([]bool, 0)
+	var colNameBuilder strings.Builder
+	comma := ""
+	for _, def := range defs {
+		if col, ok := def.(*tree.ColumnTableDef); ok {
+			if !col.Computed.Computed {
+				colTypes = append(colTypes, col.Type.(*types.T))
+				if col.Nullable.Nullability == tree.Null {
+					nullable = append(nullable, true)
+				} else {
+					nullable = append(nullable, false)
+				}
+				colNameBuilder.WriteString(comma)
+				colNameBuilder.WriteString(col.Name.String())
+				comma = ", "
+			}
+		}
+	}
+	// TODO (Butler): deal with user defined types
+	var (
+		success int
+		fail    int
+	)
+	maxTries := numRows * 1000
+	for success < numRows {
+		var valBuilder strings.Builder
+		valBuilder.WriteString("(")
+		comma := ""
+		for j := 0; j < len(colTypes); j++ {
+			valBuilder.WriteString(comma)
+			d := RandDatum(rng, colTypes[j], nullable[j])
+
+			if colTypes[j].Family() == types.OidFamily {
+				d = tree.NewDOid(tree.DInt(rand.Intn(2)))
+			}
+			/*
+				RandDatum generates bad int2's
```

Comment:
RandDatum occasionally produces int2's that cant get inserted into the db. I think (should do more testing) that the  problematic int2's are too big or null.